app/scripts/modules/appengine/src/serverGroup/configure/wizard/configFiles.component.ts

```diff
@@ -23,11 +24,13 @@ class ConfigArtifact implements IArtifactAccountPair {
   public delegate: NgAppengineConfigArtifactDelegate;
   public id: string;
   public account: string;
+  public artifact?: IArtifact;
 
-  constructor($scope: IScope, artifact = { id: '', account: '' }) {
+  constructor($scope: IScope, pair: IArtifactAccountPair = { id: '', account: '' }) {
     const unserializable = { configurable: false, enumerable: false, writable: false };
-    this.id = artifact.id;
-    this.account = artifact.account;
+    this.id = null;
```

Comment:
Why are we dropping the id instead of setting it to `pair.id`?

Reply:
Oh... when the json payload gets to orca if the artifact resolver see’s that it has an ID set it will search for it in the “resolvedArtifacts” section, which doesn’t exist. It’s a different pattern that wasn’t adhered to by the stage. Setting the id to null guarantees that it’s resolved directly on the artifact